upa/contracts/UpaVerifier.sol

```diff
@@ -132,6 +133,14 @@ contract UpaVerifier is
     /// Gas per transaction.
     uint256 private constant GAS_PER_TRANSACTION = 21000;
 
+    /// Max fixed reimbursement per challenge
+    uint256 private constant MAX_FIXED_REIMBURSEMENT = (1 << 64);
```

Comment:
I'd suggest increasing this.  This is about 18ETH, which is definitely a lot, but maybe not quite far enough outside the bounds of possibility for comfort.
I think the goal is to avoid the overflow, so lets' use 1 << 128.